functions/Remove-DbaDbSynonym.ps1

```diff
@@ -0,0 +1,161 @@
+
+    .EXAMPLE
+        PS C:\> Remove-DbaDbSynonym -SqlInstance localhost, sql2016 -Database db1, db2 -Synonym synonym1, synonym2, synonym3
+
+        Removes synonym1, synonym2, synonym3 from db1 and db2 on the local and sql2016 SQL Server instances
+
+    .EXAMPLE
+        PS C:\> $servers = Get-Content C:\servers.txt
+        PS C:\> $servers | Remove-DbaDbSynonym -Database db1, db2 -Synonym synonym1
+
+        Removes synonym1 from db1 and db2 on the servers in C:\servers.txt
+
+    .EXAMPLE
+        PS C:\> $synonyms = Get-DbaDbSynonym -SqlInstance localhost, sql2016 -Database db1, db2 -Synonym synonym1, synonym2, synonym3
+        PS C:\> $synonyms | Remove-DbaDbSynonym
+
+        Removes synonym1, synonym2, synonym3 from db1 and db2 on the local and sql2016 SQL Server instances
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [parameter(ValueFromPipeline)]
+        [DbaInstance[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [string[]]$ExcludeDatabase,
+        [string[]]$Schema,
+        [string[]]$ExcludeSchema,
+        [string[]]$Synonym,
+        [string[]]$ExcludeSynonym,
+        [parameter(ValueFromPipeline)]
+        [object[]]$InputObject,
+        [switch]$EnableException
+    )
+
+    process {
+        if (-not $InputObject -and -not $SqlInstance) {
+            Stop-Function -Message "You must pipe in a synonym, database, or server or specify a SqlInstance"
+            return
+        }
+
+        if ($SqlInstance) {
+            $InputObject = $SqlInstance
+        }
+
+        foreach ($input in $InputObject) {
+            $inputType = $input.GetType().FullName
+            switch ($inputType) {
+                'Sqlcollaborative.Dbatools.Parameter.DbaInstanceParameter' {
+                    Write-Message -Level Verbose -Message "Processing DbaInstanceParameter through InputObject"
+                    $dbSynonyms = Get-DbaDBSynonym -SqlInstance $input -SqlCredential $SqlCredential -Database $Database -ExcludeDatabase $ExcludeDatabase -Schema $Schema -ExcludeSchema $ExcludeSchema -Synonym $Synonym -ExcludeSynonym $ExcludeSynonym
+                }
+                'Microsoft.SqlServer.Management.Smo.Server' {
+                    Write-Message -Level Verbose -Message "Processing Server through InputObject"
+                    $dbSynonyms = Get-DbaDBSynonym -SqlInstance $input -SqlCredential $SqlCredential -Database $Database -ExcludeDatabase $ExcludeDatabase -Schema $Schema -ExcludeSchema $ExcludeSchema -Synonym $Synonym -ExcludeSynonym $ExcludeSynonym
+                }
+                'Microsoft.SqlServer.Management.Smo.Database' {
+                    Write-Message -Level Verbose -Message "Processing Database through InputObject"
+                    $dbSynonyms = $input | Get-DbaDBSynonym -ExcludeDatabase $ExcludeDatabase -Schema $Schema -ExcludeSchema $ExcludeSchema -Synonym $Synonym -ExcludeSynonym $ExcludeSynonym
```

Comment:
Casing should be corrected on public function reference `Get-DbaDbSynonym`

Reply:
As well this can be more easily read using `Get-DBaDbSynonym -InputObject $input`

Reply:
Cool, cool. Will have a look. It's almost 100% based on existing *-DbaDbRole

Reply:
This should be fixed now.